triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala

```diff
@@ -44,12 +46,44 @@ object TriggerServiceFixture {
       aesf: ExecutionSequencerFactory,
       ec: ExecutionContext): Future[A] = {
 
+    // Launch a toxiproxy instance . Wait on it to be ready to accept
+    // connections. Note that it binds to a hard-coded port (8474) here.
+    val toxiProxy =
+      Process(BazelRunfiles.rlocation("external/toxiproxy_nix/bin/toxiproxy-cmd")).run()
+    var retries = 0
+    var scanning = true
+    while (scanning) {
+      try {
+        println("Waiting for Toxiproxy...")
+        val channel = new java.net.Socket("localhost", 8474)
+        channel.close()
+        scanning = false
+      } catch {
+        case _: Throwable =>
+          retries = retries + 1
+          if (retries >= 3) {
+            println("Toxiproxy failed to start. Terminating with errors.")
+            System.exit(1) // Give up.
+          } else {
+            Thread.sleep(2000)
+          }
+      }
+    }
+
+    // The sandbox port is also hard-coded.
+    val client = new ToxiproxyClient("localhost", 8474);
+    val sandboxProxy: Proxy =
+      client.createProxy("sandbox", "localhost:6866", "localhost:6865")
```

Comment:
Why do we need to hardcode the sandbox port? Can't we use the `port` value below?

Reply:
We might be able to do dynamic port configuration by reordering things. I don't know yet!

Reply:
But of course, you observe that the proxy needs the sandbox port naturally.

Reply:
Looking at it again, I am convinced we can reorder the program so the sandbox does dynamic port allocation. I don't see that there's any particular merit in it though if the Toxiproxy doesn't support dynamic port allocation (as noted later in this ticket - it doesn't seem to).

Reply:
It seems like better practice to keep it dynamic where practical (like here) and later try to give a port number to the toxiproxy client which we find to be free.